### PR TITLE
Update azeventhubs to latest go-amqp

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/messaging/azeventhubs/internal/amqp_fakes.go [hunk 1]
```diff
@@ -71,10 +71,10 @@ func (ns *FakeNSForPartClient) NewAMQPSession(ctx context.Context) (amqpwrap.AMQ
 
 func (sess *FakeAMQPSession) NewReceiver(ctx context.Context, source string, opts *amqp.ReceiverOptions) (amqpwrap.AMQPReceiverCloser, error) {
 	sess.NS.NewReceiverCalled++
-	sess.NS.Receiver.ManualCreditsSetFromOptions = opts.ManualCredits
+	sess.NS.Receiver.ManualCreditsSetFromOptions = opts.Credit == -1
 	sess.NS.Receiver.CreditsSetFromOptions = opts.Credit
 
-	if !opts.ManualCredits {
+	if opts.Credit > -1 {
```
Comment (jhendrixMSFT) [inline sdk/messaging/azeventhubs/internal/amqp_fakes.go:77]: Hmm I think this should be `> 0`.

**Review by richardpark-msft — COMMENTED** [review]
Looks great!

**Review by richardpark-msft — APPROVED** [review]

**Pull request merged**